### PR TITLE
sst_importer: disable validate sst when length and crc were not both set

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/br/issues/163

Problem Summary:
set length in SSTMeta will trigger `validate_sst_for_ingestion` if we don't set crc32 if will failed.

### What is changed and how it works?
trigger `validate_sst_for_ingestion` when length and crc32 were both set.

### Related changes
- Need to cherry-pick to the release-4.0, release-3.1

### Check List 

Tests 

- Unit test

### CODE REVIEW

File: components/sst_importer/src/sst_importer.rs [hunk 1]
```diff
@@ -400,7 +400,7 @@ impl ImportDir {
         engine.prepare_sst_for_ingestion(&path.save, &path.clone)?;
         let length = meta.get_length();
         let crc32 = meta.get_crc32();
-        if length != 0 || crc32 != 0 {
+        if length != 0 && crc32 != 0 {
             // we only validate if the length and CRC32 are explicitly provided.
             engine.validate_sst_for_ingestion(cf, &path.clone, length, crc32)?;
```
Comment (overvenus) [inline components/sst_importer/src/sst_importer.rs]: We should validate sst file size.

Comment (3pointer) [reply components/sst_importer/src/sst_importer.rs]: okay, check filesize and crc32 separately, when filesize or crc32 > 0

File: tests/integrations/import/sst_service.rs [hunk 2]
```diff
@@ -108,6 +108,13 @@ fn test_ingest_sst() {
     let resp = import.ingest(&ingest).unwrap();
     assert!(!resp.has_error());
 
+    // Set crc32 == 0 and length != 0
+    meta.set_crc32(0);
+    meta.set_length(data.len() as u64);
+    ingest.set_sst(meta.clone());
+    let resp = import.ingest(&ingest).unwrap();
+    assert!(!resp.has_error());
```
Comment (kennytm) [inline tests/integrations/import/sst_service.rs]:
```
[2020-03-25T10:02:13.351Z] test import::sst_service::test_ingest_sst ... thread 'apply-1' panicked at '[region 1] 1 ingest uuid: 50F9339230554764AD40201F4348BE93 range { start: 00 end: 64 } length: 1634 cf_name: "default" region_id: 1 region_epoch { conf_ver: 1 version: 1 }: EngineTraits(Other("[components/engine_rocks/src/import.rs:22]: Storage Engine read metadata from /tmp/test_cluster4qkiE7/db/import-sst/50f93392-3055-4764-ad40-201f4348be93_1_1_1.sst: Os { code: 2, kind: NotFound, message: \"No such file or directory\" }"))', components/raftstore/src/store/fsm/apply.rs:1410:17
```

The SST file is automatically deleted after ingestion ended.

Comment (3pointer) [reply tests/integrations/import/sst_service.rs]: oh I see

**Pull request merged**